sdk/core/azure-core-test/src/main/java/com/azure/core/test/InterceptorManager.java

```diff
@@ -62,6 +70,12 @@ public class InterceptorManager implements AutoCloseable {
     // Stores a map of all the HTTP properties in a session
     // A state machine ensuring a test is always reset before another one is setup
     private final RecordedData recordedData;
+    private final boolean enableTestProxy;
+    private TestProxyRecordPolicy testProxyRecordPolicy;
+    private TestProxyPlaybackClient testProxyPlaybackClient;
+    private final Queue<String> proxyVariableQueue = new LinkedList<>();
```

Comment:
Yes, the order of insertion should matter here. 
https://github.com/billwert/azure-sdk-for-java/blob/test-recorder/sdk/core/azure-core-test/src/test/resources/session-records/TestProxyTests.testPlaybackWithRedaction.json#L26

Reply:
Yeah. This is the storage for `testResourceNamer`. Because of how the test proxy works we need to batch them up and send them all when we stop recording. Ordering does matter so that replay of random names works properly.